### ISSUE TITLE
Satisfaction: Handle byproduct only parts

### ISSUE THREAD

**Issue by Maelstromeous** [satisfactory-factories/application#220]
Fix Production is showing up in Satisfaction for a byproduct. Clicking it however does nothing, presumably because we don't know which product to increase to increase the byproduct.

Maybe we need to inform the user better saying "Increase production of byproduct" or something like that rather than trying to auto-fix something.

https://satisfactory-factories.app/share/beautiful-important-man (love the link :D)

https://github.com/user-attachments/assets/0fd6b47c-60a0-4f16-a4ef-bddb06ef5db3

**Comment by Maelstromeous** [comment]
Upgraded priority, it's quite jarring.

**Comment by samsmithnz** [comment]
I also see the inverse issue with the `Trim` button. 
To replicate, create AL Casings, and add all the products. The SIlica byproduct needs to be subsidized by a production line, causing a surplus the factory can't reduce. Pressing the `Trim` button does nothing.

![Image](https://github.com/user-attachments/assets/1f75fd8e-c58a-49c5-9fa0-166ddb3cd01b)

**Comment by Maelstromeous** [comment]
For now, I'm going to add a "fake" button with a tooltip. Creating a button to directly fix this issue by creating a product with the direct byproduct as a product is currently problematic due to https://github.com/satisfactory-factories/application/issues/95.

**Comment by Maelstromeous** [comment]
<img width="1299" alt="Screenshot 2025-01-10 at 14 51 26" src="https://github.com/user-attachments/assets/aa432030-1b09-4197-9c0c-7674e02d8cfc" />

My resolution. @samsmithnz what you think?

**Comment by samsmithnz** [comment]
Makes sense. I do think we can solve this automatically - but this is a good short term solution. What does this look like if the user also has some heavy oil residue manually added?

**Comment by Maelstromeous** [comment]
![Screenshot 2025-01-10 at 15 07 56](https://github.com/user-attachments/assets/01c24a2b-512b-41a1-9763-963d4346c99a)
It converts into fix product

**Comment by samsmithnz** [comment]
perfect. Looks good to me!
